Generator does not support NSArray (or arrays) for CoreImage filters

### Steps to Reproduce

1. Uncomment `CIVector []` from `coreimage.cs`
2. Build

### Expected Behavior

Generated getter and setter

### Actual Behavior

```
error BI1018: bgen: Unimplemented CoreImage property type CIVector[]
```

### Environment

```
xamarin-macios/xcode10
```

### Build Logs

```
GEN      [tvos] generated_sources
error BI1018: bgen: Unimplemented CoreImage property type CIVector[]
make[1]: *** [build/tvos/tvos/generated_sources] Error 1
make[1]: *** Waiting for unfinished jobs....
error BI1018: bgen: Unimplemented CoreImage property type CIVector[]
make[1]: *** [build/ios/native/generated_sources] Error 1
error BI1018: bgen: Unimplemented CoreImage property type CIVector[]
make[1]: *** [build/mac/mobile/generated-sources] Error 1
error BI1018: bgen: Unimplemented CoreImage property type CIVector[]
make[1]: *** [build/mac/full/generated-sources] Error 1
make: *** [all-recurse] Error 1
```

--- a/src/coreimage.cs
+++ b/src/coreimage.cs
@@ -5563,9 +5563,8 @@ namespace CoreImage {
 	[Mac (10,14, onlyOn64: true)]
 	[BaseType (typeof (CIFilter))]
 	interface CIMeshGenerator {
-		// https://github.com/xamarin/xamarin-macios/issues/4226
-		//[CoreImageFilterProperty ("inputMesh")]
-		//CIVector [] Mesh { get; set; }
+		[CoreImageFilterProperty ("inputMesh")]
+		CIVector [] Mesh { get; set; }
 		[CoreImageFilterProperty ("inputWidth")]
 		float Width { get; set; }
 		[CoreImageFilterProperty ("inputColor")]

--- a/src/generator-filters.cs
+++ b/src/generator-filters.cs
@@ -252,8 +252,15 @@ public partial class Generator {
 			indent--;
 			break;
 		case "CIVector[]":
-			print ("var array = NSArray.FromNSObjects (value);");
-			print ($"SetHandle (\"{propertyName}\", array.GetHandle ());");
+			print ("if (value == null) {");
+			indent++;
+			print ($"SetHandle (\"{propertyName}\", IntPtr.Zero);");
+			indent--;
+			print ("} else {");
+			indent++;
+			print ($"SetHandle (\"{propertyName}\", NSArray.FromNSObjects (value).GetHandle ());");
+			indent--;
+			print ("}");
 			break;
 		default:
 			throw new BindingException (1018, true, "Unimplemented CoreImage property type {0}", propertyType);

--- a/src/generator-filters.cs
+++ b/src/generator-filters.cs
@@ -199,6 +199,10 @@ public partial class Generator {
 			// NSString should not be added - it should be bound as a string
 			print ("return (string) (ValueForKey (\"{0}\") as NSString);", propertyName);
 			break;
+		case "CIVector[]":
+			print ($"var handle = GetHandle (\"{propertyName}\");");
+			print ("return NSArray.ArrayFromHandle<CIVector> (handle);");
+			break;
 		default:
 			throw new BindingException (1018, true, "Unimplemented CoreImage property type {0}", propertyType);
 		}
@@ -246,6 +250,10 @@ public partial class Generator {
 			indent++;
 			print ("SetValue (\"{0}\", ns);", propertyName);
 			indent--;
+			break;
+		case "CIVector[]":
+			print ("var array = NSArray.FromNSObjects (value);");
+			print ($"SetHandle (\"{propertyName}\", array.GetHandle ());");
 			break;
 		default:
 			throw new BindingException (1018, true, "Unimplemented CoreImage property type {0}", propertyType);

--- a/src/generator-filters.cs
+++ b/src/generator-filters.cs
@@ -258,7 +258,10 @@ public partial class Generator {
 			indent--;
 			print ("} else {");
 			indent++;
-			print ($"SetHandle (\"{propertyName}\", NSArray.FromNSObjects (value).GetHandle ());");
+			print ("using (var array = NSArray.FromNSObjects (value))");
+			indent++;
+			print ($"SetHandle (\"{propertyName}\", array.GetHandle ());");
+			indent--;
 			indent--;
 			print ("}");
 			break;

--- a/tests/monotouch-test/CoreImage/FilterTest.cs
+++ b/tests/monotouch-test/CoreImage/FilterTest.cs
@@ -153,6 +153,18 @@ namespace MonoTouchFixtures.CoreImage {
 				}
 			}
 		}
+
+		[Test]
+		public void CIVectorArray ()
+		{
+			TestRuntime.AssertXcodeVersion (10, 0);
+
+			using (var f = new CIMeshGenerator ()) {
+				Assert.Null (f.Mesh, "Mesh/Null");
+				f.Mesh = new CIVector [1] { new CIVector (1) };
+				Assert.That (f.Mesh.Length, Is.EqualTo (1), "Mesh/Non-null");
+			}
+		}
 	}
 }
 

--- a/tests/monotouch-test/CoreImage/FilterTest.cs
+++ b/tests/monotouch-test/CoreImage/FilterTest.cs
@@ -163,6 +163,8 @@ namespace MonoTouchFixtures.CoreImage {
 				Assert.Null (f.Mesh, "Mesh/Null");
 				f.Mesh = new CIVector [1] { new CIVector (1) };
 				Assert.That (f.Mesh.Length, Is.EqualTo (1), "Mesh/Non-null");
+				f.Mesh = null;
+				Assert.Null (f.Mesh, "Mesh/Null/again");
 			}
 		}
 	}

